UART Driver/Core/Src/main.c

```diff
@@ -0,0 +1,307 @@
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+UART_HandleTypeDef huart2;
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+static void MX_GPIO_Init(void);
+static void MX_USART2_UART_Init(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+//Creates the fijo and foji struct
+//The reason there are 2 different gpsCoordinates structs is to minimize the size of the fijo struct
+struct gpsCoordinatesFOJI{
+	double lattitude;
+	double longtitude;
+	double altitude;
+};
+
+struct gpsCoordinatesFIJO{
+	double lattitude;
+	double longtitude;
+};
+
+struct eulerAnglesOfPlane{
+	float yaw;
+	float pitch;
+	float roll;
+};
+
+//Should be 37 bytes but because of padding its really 48 bytes
+ struct foji{
+	uint8_t start;
+	struct gpsCoordinatesFOJI gpsCoord;
+	struct eulerAnglesOfPlane anglesOfPlane;
+};
+
+//Should be 20 bytes but because of padding its really 24 bytes
+struct fijo{
+	 uint8_t start;
+	_Bool takeoffCommand;
+	_Bool qrScanFlag;
+	_Bool detectFlag;
+	struct gpsCoordinatesFIJO gpsCoord;
+};
+
+const uint8_t STARTBYTE = 0x24;
+
+void sendFOJI(){
+	//Get out from telemetry manager
+	struct foji out;
+
+	HAL_UART_Transmit(&huart2, &out, sizeof(struct foji), HAL_MAX_DELAY);
+
+}
+
+void readFIJO(){
+	//Creates byte array
+	uint8_t buf[sizeof(struct fijo)];
+
+	//Initializes starting byte
+	buf[0] = STARTBYTE;
+
+	struct fijo in;
+	in.start = STARTBYTE;
+
+	//Receives a byte to check if its the same as STARTBYTE
+	uint8_t check;
+	HAL_UART_Receive(&huart2, &check, 1, HAL_MAX_DELAY);
+
+	if(check == STARTBYTE){
+		//Receive all the message bytes in buf
+		for(uint16_t i = 1; i < sizeof(struct fijo); i++)
+		{
+			HAL_UART_Receive(&huart2, &buf[i], 1, HAL_MAX_DELAY);
+		}
+
+		//Convert the byte array into an fijo struct
+		struct fijo *ptr = (struct fijo*)buf;
```

Comment:
Not sure if we want to use "ptr" or if we want a more descriptive name.

Reply:
Ya fs I changed it to a more meaningful name